### DISCUSSION TITLE
How to smooth imported mesh?

### DISCUSSION THREAD

**Discussion by zaichenhu** [mosra/magnum#585] [Q&A] [answered]
I have imported a triangulated mesh using magnum scene importer, but it was shaded flat when rendering. if there is any api to solve this problem?

**Comment by zaichenhu** [comment]
change the compile flag to MeshTools::CompileFlag::GenerateSmoothNormals could work , sorry to disturb...

**Comment by mosra** [comment, marked as answer]
Yup, that's the flag :) 

To answer fully -- smooth normal generation relies on the index buffer, so if the mesh is not indexed, you'll still get flat normals. To fix that, perform [MeshTools::removeDuplicates()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1MeshTools.html#ac39584845d0291262e8276e05d53b22e) first, which will generate the index buffer.
